### PR TITLE
Update javadoc @since for API additions

### PR DESCRIPTION
**Pull request by MarkEWaite**
## Update javadoc `@since` for recent API additions

Update the `@since` javadoc comments for recent additions to the Jenkins API.

List of commits introducing new API and the first release they went in:
* https://github.com/jenkinsci/jenkins/releases/tag/jenkins-2.323
  - https://github.com/jenkinsci/jenkins/commit/ca67a716388a326e2af4477dc001464bd6da8f96
  - https://github.com/jenkinsci/jenkins/commit/cc1966388287e4e49f6de9cfd165afcbdd17b6da
* https://github.com/jenkinsci/jenkins/releases/tag/jenkins-2.324
  - https://github.com/jenkinsci/jenkins/commit/e4d74d3691e410b43d9225dd95a3c805d97a1ce6
* https://github.com/jenkinsci/jenkins/releases/tag/jenkins-2.325
  - https://github.com/jenkinsci/jenkins/commit/c97b55f9f18d6e20244644ae7a688707bbb9cf90
  - https://github.com/jenkinsci/jenkins/commit/59cc04c28a2229275f4ea0be82765ef83559b821
  - https://github.com/jenkinsci/jenkins/commit/b4a622922c99f756c9b2e997845d4dd68e550386
  - https://github.com/jenkinsci/jenkins/commit/b6e148e6ef43e302debf27edb46a557b932beca8
* https://github.com/jenkinsci/jenkins/releases/tag/jenkins-2.335
  - https://github.com/jenkinsci/jenkins/commit/81072df4f7f3c0ecef0d5ab20100a87cb0acce58
  - https://github.com/jenkinsci/jenkins/commit/7915d822cb84a6a17568dfdc55cfdb21c49dc97f

### Proposed changelog entries

* N/A

### Proposed upgrade guidelines

N/A

### Submitter checklist

- [x] (If applicable) Jira issue is well described
- [x] Changelog entries and upgrade guidelines are appropriate for the audience affected by the change (users or developer, depending on the change) and are in the imperative mood. [Examples](https://github.com/jenkins-infra/jenkins.io/blob/master/content/_data/changelogs/weekly.yml)
  * Fill-in the `Proposed changelog entries` section only if there are breaking changes or other changes which may require extra steps from users during the upgrade
- [x] Appropriate autotests or explanation to why this change has no tests
- [x] New public classes, fields, and methods are annotated with `@Restricted` or have `@since TODO` Javadoc, as appropriate. 
- [x] For dependency updates: links to external changelogs and, if possible, full diffs

### Maintainer checklist

Before the changes are marked as `ready-for-merge`: 

- [ ] There are at least 2 approvals for the pull request and no outstanding requests for change
- [ ] Conversations in the pull request are over OR it is explicit that a reviewer does not block the change
- [ ] Changelog entries in the PR title and/or `Proposed changelog entries` are accurate, human-readable, and in the imperative mood
- [ ] Proper changelog labels are set so that the changelog can be generated automatically
- [ ] If the change needs additional upgrade steps from users, `upgrade-guide-needed` label is set and there is a `Proposed upgrade guidelines` section in the PR title. ([example](https://github.com/jenkinsci/jenkins/pull/4387))
- [ ] If it would make sense to backport the change to LTS, a Jira issue must exist, be a _Bug_ or _Improvement_, and be labeled as `lts-candidate` to be considered (see [query](https://issues.jenkins-ci.org/issues/?filter=12146)).

### CODE REVIEW

**Comment by daniel-beck** [comment]
BTW, the output of `update-at-since.sh` makes a good PR comment.

**Review by basil — APPROVED** [review]
This PR is now ready for merge. We will merge it after approximately 24 hours if there is no negative feedback. Please see the [merge process documentation](https://github.com/jenkinsci/jenkins/blob/master/docs/MAINTAINERS.adoc#merge-process) for more information about the merge process. Thanks!

File: core/src/main/java/hudson/lifecycle/Lifecycle.java
Comment (daniel-beck) [inline core/src/main/java/hudson/lifecycle/Lifecycle.java:307]: This one is wrong. Line was added (badly) in https://github.com/MarkEWaite/jenkins/commit/7915d822cb84a6a17568dfdc55cfdb21c49dc97f.

Comment (MarkEWaite) [reply core/src/main/java/hudson/lifecycle/Lifecycle.java:307]: Apologies if I made a mistake there, but I'm not sure of the mistake I made.  Can you clarify for me so that I can correct it?

The API `onExtendTimeout(long timeout, @NonNull TimeUnit unit)` was added in commit 3866e93a364 as far as I can tell.  The first release tag to include that was [2.335](https://github.com/jenkinsci/jenkins/releases/tag/jenkins-2.335) as part of jenkinsci/jenkins#6237.

Comment (daniel-beck) [reply core/src/main/java/hudson/lifecycle/Lifecycle.java:307]: Ah, I missed that these weren't released earlier, just saw that a different PR was identified.

**Pull request merged**
